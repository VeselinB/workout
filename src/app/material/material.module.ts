import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DragDropModule } from '@angular/cdk/drag-drop';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatDialogModule } from "@angular/material/dialog";
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatInputModule } from '@angular/material/input';
import { MatSelectModule } from '@angular/material/select';
import { MatSidenavModule } from '@angular/material/sidenav';

import { MatProgressSpinnerModule } from '@angular/material/progress-spinner';
import { MatIconModule } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { MatExpansionModule } from '@angular/material/expansion';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatMenuModule } from '@angular/material/menu';
import { MatTabsModule } from '@angular/material/tabs';
import { MatTooltipModule } from '@angular/material/tooltip';
import { MatNativeDateModule } from '@angular/material/core';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatTableModule } from '@angular/material/table';
import { MatDividerModule } from '@angular/material/divider';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { MatChipsModule } from '@angular/material/chips';
import { MatAutocompleteModule } from '@angular/material/autocomplete';
import { MatBadgeModule } from '@angular/material/badge';
import { MatSnackBarModule } from '@angular/material/snack-bar';
import { MatSortModule } from '@angular/material/sort';
import {MatRadioModule} from '@angular/material/radio'; 









const MODULES = [
  MatSortModule,
  MatRadioModule,
  MatSortModule,
  MatSnackBarModule,
  MatBadgeModule,
  MatAutocompleteModule,
  MatChipsModule,
  MatProgressBarModule,
  MatDividerModule,
  MatTableModule,
  MatPaginatorModule,
  MatNativeDateModule,
  MatTooltipModule,
  MatTabsModule,
  MatMenuModule,
  MatDatepickerModule,
  MatExpansionModule,
  MatListModule,
  MatIconModule,
  MatProgressSpinnerModule,
  MatToolbarModule,
  MatSidenavModule,
  MatSelectModule,
  MatInputModule,
  DragDropModule,
  MatCardModule,
  MatButtonModule,
  MatDialogModule,
  MatFormFieldModule
]

@NgModule({
  declarations: [],

  imports: [CommonModule, ...MODULES],
  exports: [...MODULES],

})
export class MaterialModule { }