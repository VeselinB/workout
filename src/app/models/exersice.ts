export interface Exersice {
    id: string;
    title: string;
    status: Boolean;
}

export class ExersiceClass {
    id: string;
    title: string;
    status: Boolean;

    constructor(id: string, title: string) {
        this.id = id;
        this.title = title;
        this.status = false;
   
    }
}