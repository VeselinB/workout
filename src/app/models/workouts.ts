import { Exersice } from "./exersice";

export interface WorkOut {
    exersices: Exersice[];
    status: Boolean;
}