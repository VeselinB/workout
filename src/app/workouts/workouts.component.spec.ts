import { ComponentFixture, TestBed } from '@angular/core/testing';
import { reducer } from './Workout.reducer';
import { StoreModule } from '@ngrx/store';
import { WorkoutsComponent } from './Workouts.component';
import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
import { of } from 'rxjs';
import { MaterialModule } from '../material/material.module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { HttpClientModule } from '@angular/common/http';
import { ExersiceClass } from '../models/exersice';

declare var require: any
// const { toMatchImageSnapshot } = require('jest-image-snapshot');
// const puppeteer = require('puppeteer');
// expect.extend({ toMatchImageSnapshot });
export class MatDialogMock {

  open() {
    return {
      afterClosed: () => of({ action: true })
    };
  }
}

describe('WorkoutsComponent', () => {
  let browser;
  let component: WorkoutsComponent;
  let fixture: ComponentFixture<WorkoutsComponent>;

  beforeEach(async () => {
   // browser = await puppeteer.launch();
    await TestBed.configureTestingModule({
      imports: [
        MaterialModule, BrowserAnimationsModule,
        HttpClientModule,
        StoreModule.forRoot({
          Workout: reducer,
        })
      ],
      declarations: [WorkoutsComponent],
      providers: [{ provide: MatDialog, useClass: MatDialogMock }]
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(WorkoutsComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it("should create new exercise", () => {
    let exercise = new ExersiceClass("id", "Title");
    expect(typeof exercise).toBe('object');
    expect(exercise.title).toBe("Title");
    expect(exercise.id).toBe("id");
    expect(exercise.status).toBe(false);
  });

  it('render correctly',async  () => {
    // const page = await browser.newPage();
    // await page.goto('https://localhost:3000');
    // const image = await page.screenshot();
    const element: HTMLElement = fixture.nativeElement;
    component.activeWorkouts = [{
      id: "1",
      exersices: {},
      status: false,
      title: "Test",
    }];
    fixture.detectChanges();
   expect(fixture).toMatchSnapshot();
   expect(element.textContent).toContain("exercise");
  });
});


