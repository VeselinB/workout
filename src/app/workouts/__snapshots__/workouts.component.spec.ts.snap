// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutsComponent render correctly 1`] = `
<app-workouts
  Object={[Function Function]}
  activeWorkOuts={[Function Array]}
  callAPIDialog={[Function TemplateRef_]}
  completedWorkOuts={[Function Array]}
  dialog={[Function MatDialogMock]}
  disabled={[Function Boolean]}
  name={[Function Object]}
  newWork={[Function WorkOutClass]}
  store={[Function Store]}
>
  <div
    style="margin-left: 20px; margin-right: 20px;"
  >
    <h2>
      WorkOuts
    </h2>
    <button
      class="mat-focus-indicator mat-raised-button mat-button-base mat-primary"
      color="primary"
      mat-raised-button=""
      style="margin-bottom: 20px;"
    >
      <span
        class="mat-button-wrapper"
      >
        <mat-icon
          aria-hidden="true"
          class="mat-icon notranslate material-icons mat-icon-no-color"
          data-mat-icon-type="font"
          role="img"
        >
          add
        </mat-icon>
        New workout 
      </span>
      <span
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <span
        class="mat-button-focus-overlay"
      />
    </button>
    <div
      style="display: flex; flex-direction: row; width: 100%;"
    >
      <mat-card
        class="mat-card mat-focus-indicator"
        style="width: 100%; margin-right: 6px;"
      >
        <mat-card-title
          class="mat-card-title"
        >
          Active
        </mat-card-title>
        <mat-card
          class="mat-card mat-focus-indicator"
          style="margin-bottom: 8px;"
        >
          <mat-card-subtitle
            class="mat-card-subtitle"
          >
            Test
          </mat-card-subtitle>
          <mat-selection-list
            aria-disabled="false"
            aria-multiselectable="true"
            class="mat-selection-list mat-list-base"
            role="listbox"
            tabindex="-1"
          >
          </mat-selection-list>
          <p />
          <mat-card-actions
            class="mat-card-actions"
          >
            <button
              class="mat-focus-indicator mat-raised-button mat-button-base mat-warn"
              color="warn"
              mat-raised-button=""
            >
              <span
                class="mat-button-wrapper"
              >
                Delete
              </span>
              <span
                class="mat-button-ripple mat-ripple"
                matripple=""
              />
              <span
                class="mat-button-focus-overlay"
              />
            </button>
            <button
              class="mat-focus-indicator mat-raised-button mat-button-base mat-primary"
              color="primary"
              mat-raised-button=""
            >
              <span
                class="mat-button-wrapper"
              >
                <mat-icon
                  aria-hidden="true"
                  class="mat-icon notranslate material-icons mat-icon-no-color"
                  data-mat-icon-type="font"
                  role="img"
                >
                  add
                </mat-icon>
                New exercise 
              </span>
              <span
                class="mat-button-ripple mat-ripple"
                matripple=""
              />
              <span
                class="mat-button-focus-overlay"
              />
            </button>
          </mat-card-actions>
        </mat-card>
        <br />
      </mat-card>
      <br />
      <mat-card
        class="mat-card mat-focus-indicator"
        style="width: 100%;"
      >
        <mat-card-title
          class="mat-card-title"
        >
          Completed
        </mat-card-title>
        <br />
      </mat-card>
      <br />
    </div>
  </div>
</app-workouts>
`;
